web-ui.py

```diff
@@ -491,6 +494,62 @@ def save_config():
         })
 
 
+@app.route('/get_beatmap_info', methods=['GET'])
+def get_beatmap_info():
+    """Get auto-fill information from a beatmap file."""
+    try:
+        beatmap_path = request.args.get('beatmap_path')
+        if not beatmap_path:
+            return jsonify({'success': False, 'error': 'No beatmap path provided'})
+
+        # Use the utility function from inference.py
+        result = get_autofill_paths_from_beatmap(beatmap_path)
+
+        return jsonify({
+            'success': result['error'] is None,
+            'data': {
+                'audio_path': result['audio_path'],
+                'output_path': result['output_path'],
+                'audio_filename': result['audio_filename'],
+                'audio_exists': result['audio_exists']
+            },
+            'error': result['error']
+        })
+
+    except Exception as e:
+        print(f"Error in get_beatmap_info: {e}")
+        return jsonify({
+            'success': False,
+            'error': f'Failed to get beatmap info: {str(e)}'
+        })
+
+
+@app.route('/get_directory_from_path', methods=['GET'])
+def get_directory_from_path():
```

Comment:
why is this a route in web-ui? Getting the parent directory can easily be done in JavaScript